### PR TITLE
Update packages and fix models refs

### PR DESCRIPTION
**Pull request by fj-sanchez**
A few changes have been included here:

* Updated httpx, fastapi and pydantic to latest versions.
* Updated deprecated methods/classes from httpx and pydantic.
* Added automatic update of forward references for pydantic models.
* Docker images will be run as current user instead of root.

### CODE REVIEW

File: openapi-python-templates/model.mustache [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
 {{#model}}
 class {{classname}}(BaseModel):
 {{#vars}}
 {{#isEnum}}
-    {{name}}: 'Literal[{{#allowableValues}}{{#values}}"{{{this}}}"{{^-last}}, {{/-last}}{{/values}}{{/allowableValues}}]' = Schema({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: ignore
+    {{name}}: Literal[{{#allowableValues}}{{#values}}"{{{this}}}"{{^-last}}, {{/-last}}{{/values}}{{/allowableValues}}] = Field({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: inore
 {{/isEnum}}
 {{^isEnum}}
-    {{name}}: "{{^required}}Optional[{{/required}}{{>_dataTypeModel}}{{^required}}]{{/required}}" = Schema({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: ignore
+    {{name}}: {{^required}}Optional[{{/required}}{{>_dataTypeModel}}{{^required}}]{{/required}} = Field({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: ignore
```
Comment (fj-sanchez) [inline openapi-python-templates/model.mustache]: I think that maybe I should revert this change... it works well with python 3.7+ because of the support of deferred annotations, but this cannot be imported in python 3.6 for example, which makes the client incompatible with anything below 3.7... What do you think?

File: openapi-python-templates/model.mustache [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 {{#model}}
 class {{classname}}(BaseModel):
 {{#vars}}
 {{#isEnum}}
-    {{name}}: 'Literal[{{#allowableValues}}{{#values}}"{{{this}}}"{{^-last}}, {{/-last}}{{/values}}{{/allowableValues}}]' = Schema({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: ignore
+    {{name}}: "Literal[{{#allowableValues}}{{#values}}'{{{this}}}'{{^-last}}, {{/-last}}{{/values}}{{/allowableValues}}]" = Field({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: inore
 {{/isEnum}}
 {{^isEnum}}
-    {{name}}: "{{^required}}Optional[{{/required}}{{>_dataTypeModel}}{{^required}}]{{/required}}" = Schema({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: ignore
+    {{name}}: "{{^required}}Optional[{{/required}}{{>_dataTypeModel}}{{^required}}]{{/required}}" = Field({{#required}}...{{/required}}{{^required}}None{{/required}}, alias="{{baseName}}")  # type: ignore
```
Comment (dmontagu) [inline openapi-python-templates/model.mustache]: I think you can get rid of the `# type: ignore` now

File: openapi-python-templates/__init__package.mustache [hunk 3]
```diff
@@ -1 +1,10 @@
+import inspect
+
+from @IMPORT_NAME@ import models  # noqa F401
 from @IMPORT_NAME@.api_client import ApiClient, AsyncApis, SyncApis  # noqa F401
+
+
+for model in inspect.getmembers(models, inspect.isclass):
+	if model[1].__module__ == "@IMPORT_NAME@.models":
+		model_class = model[1]
+		model_class.update_forward_refs()
```
Comment (dmontagu) [inline openapi-python-templates/__init__package.mustache]: Could you add a newline at the end of file?

File: openapi-python-templates/model.mustache [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 {{/isEnum}}
 {{/vars}}
 {{/model}}
-{{/models}}
+{{/models}}
```
Comment (dmontagu) [inline openapi-python-templates/model.mustache]: Also here, can we keep the newline at end of file that used to be there

File: openapi-python-templates/model.mustache (hunk 1 above)
Comment (dmontagu) [reply openapi-python-templates/model.mustache]: I think we should keep with the `"`s for now; compatibility with 3.6 is still nice to offer, and you need them either way if you don't add `from __future__ import annotations` at the top.

**Comment by dmontagu** [comment]
This is great! Thanks so much for all the updates. The run-as-user for docker is also nice. If you could just make the minor changes noted above I'll merge this.

-----

Regarding the `Dict[str, ...]` change -- I'm not sure what the fastapi-generated OpenAPI spec looks like if you annotate a value as `Dict[int, int]` but I could imagine the `Dict[str, int]` being mismatched in that case. Either way, I think the str-valued-keys case is going to be by far the most common, so I think it makes sense to make this change and fix it in the future if someone requests it (or we find an easy way).

File: openapi-python-templates/model.mustache (hunk 2 above)
Comment (fj-sanchez) [reply openapi-python-templates/model.mustache]: Every type is a string AFAIK.

File: openapi-python-templates/__init__package.mustache (hunk 3 above)
Comment (fj-sanchez) [reply openapi-python-templates/__init__package.mustache]: Done, Black will reformat it anyway.

File: openapi-python-templates/model.mustache (hunk 4 above)
Comment (fj-sanchez) [reply openapi-python-templates/model.mustache]: Done.
